great_expectations/rule_based_profiler/builder.py

```diff
@@ -73,7 +80,9 @@ def data_context(self) -> Optional[AbstractDataContext]:
     def set_batch_list_if_null_batch_request(
         self,
         batch_list: Optional[List[Batch]] = None,
-        batch_request: Optional[Union[BatchRequestBase, dict]] = None,
+        batch_request: Optional[
+            Union[BatchRequestBase, BlockConfigBatchRequestTypedDict, dict]
+        ] = None,
```

Comment:
```suggestion
        batch_request: BatchRequestBase | dict | BlockConfigBatchRequestTypedDict | None = None,
```

Reply:
Is there a preference for styles for typing? Is there a case when one is preferred over the other eg. for return values vs params

Reply:
I think most developers (including me) prefer the alternative union syntax -> `str | int`
https://peps.python.org/pep-0604/

The only reason I haven't been always updating the syntax is if I'm trying to reduce the number of changes in the git diff.

Also, we have a few cases where we can't use this syntax because we are using the type information at runtime (such as on a pydantic model), and because we are on python 3.8 the only way to use the new syntax is with `from __future__ import annotations`